src/client/app/components/admin/PreferencesComponent.tsx

```diff
@@ -43,6 +43,40 @@ export default function PreferencesComponent() {
 		setLocalAdminPref(cloneDeep(adminPreferences));
 	};
 
+	// small functions that determine if fields are invalid to warn the user
+	const invalidReadingFreq = () => {
+		const frequency = moment.duration(localAdminPref.defaultMeterReadingFrequency);
+		return !frequency.isValid() || frequency.asSeconds() <= 0;
+	};
+
+	// force check some localAdminPref values as numbers, they are stored as strings
+	const invalidValueLimits = () => {
+		return Number(localAdminPref.defaultMeterMinimumValue) >= Number(localAdminPref.defaultMeterMaximumValue);
+	};
+
+	const invalidDateLimits = () => {
+		const minMoment = moment(localAdminPref.defaultMeterMinimumDate);
+		const maxMoment = moment(localAdminPref.defaultMeterMaximumDate);
+		return !minMoment.isValid() || !maxMoment.isValid() || !minMoment.isBefore(maxMoment);
+	};
+
+	const invalidReadingGap = () => {
```

Comment:
These work fine but you might consider the min/max keys as in src/client/app/components/meters/CreateMeterModalComponent.tsx. They can even compare to another field and control the input value appropriately.

Reply:
Made the necessary changes